test/net-test.js

```diff
@@ -0,0 +1,2218 @@
+          assert.equal(packet.type, packets.types.PONG);
+          assert.bufferEqual(packet.nonce, nonce);
+          called = true;
+        };
+
+        await peer.handlePing(pkt);
+        assert.equal(called, true);
+      });
+    });
+
+    describe('handlePong', function() {
+      it('will not update last pong w/o challenge', async () => {
+        const peer = Peer.fromOptions({});
+        peer.challenge = null;
+        peer.lastPong = -1;
+        peer.minPing = -1;
+
+        const pkt = new packets.PongPacket();
+        await peer.handlePong(pkt);
+
+        assert.equal(peer.lastPong, -1);
+        assert.equal(peer.minPing, -1);
+      });
+
+      it('will not update last pong w/ wrong nonce', async () => {
+        const peer = Peer.fromOptions({});
+        peer.challenge = common.nonce();
+        peer.lastPong = -1;
+        peer.minPing = -1;
+
+        const pkt = new packets.PongPacket(common.nonce());
+        await peer.handlePong(pkt);
+
+        assert.equal(peer.lastPong, -1);
+        assert.equal(peer.minPing, -1);
+      });
+
+      it('will update last pong and min ping', async () => {
+        const now = Date.now();
+
+        const peer = Peer.fromOptions({});
+        const nonce = common.nonce();
+        peer.challenge = nonce;
+        peer.lastPong = -1;
+        peer.minPing = -1;
+
+        const pkt = new packets.PongPacket(nonce);
+        await peer.handlePong(pkt);
+
+        assert(peer.lastPong >= now);
+        assert(peer.minPing >= now - 1);
+      });
+    });
+
+    describe('handleSendHeaders (BIP130)', function() {
+      it('will set prefer headers', async () => {
+        const peer = Peer.fromOptions({});
+        const pkt = new packets.SendHeadersPacket();
+        await peer.handleSendHeaders(pkt);
+        assert.equal(peer.preferHeaders, true);
+      });
+    });
+
+    describe('handleFilterLoad (BIP37)', function() {
+      it('will load spv filter', async () => {
+        const peer = Peer.fromOptions({});
+        const filter = new BloomFilter();
+        const pkt = new packets.FilterLoadPacket(filter);
+        peer.handleFilterLoad(pkt);
+        assert.strictEqual(peer.spvFilter, filter);
+      });
+
+      it('will increase ban if not within constraints', async () => {
+        const peer = Peer.fromOptions({});
+        const filter = new BloomFilter();
+        const pkt = new packets.FilterLoadPacket(filter);
+
+        let called = false;
+        peer.increaseBan = (score) => {
+          assert.equal(score, 100);
+          called = true;
+        };
+
+        pkt.isWithinConstraints = () => false;
+        await peer.handleFilterLoad(pkt);
+
+        assert.equal(called, true);
+        assert.strictEqual(peer.spvFilter, null);
+      });
+    });
+
+    describe('handleFilterAdd (BIP37)', function() {
+      it('will add to spv filter', async () => {
+        const peer = Peer.fromOptions({});
+        peer.spvFilter = BloomFilter.fromRate(
```

Comment:
Do you want to test failure to `filteradd` if `filterload` has not been sent yet? (i.e. `peer.spvFilter = null`)?

Reply:
There is `if` statement for it, so it's simple:
```js
if (this.spvFilter)
  this.spvFilter.add(data);
```
So probably not in this case.

Reply:
It is a pretty short test though, may as well.